backend/controller/lmevaljob_controller_test.go

```diff
@@ -85,3 +96,687 @@ var _ = Describe("LMEvalJob Controller", func() {
+				RunAsNonRoot: &runAsNonRootUser,
+				SeccompProfile: &corev1.SeccompProfile{
+					Type: corev1.SeccompProfileTypeRuntimeDefault,
+				},
+			},
+			Volumes: []corev1.Volume{
+				{
+					Name: "shared", VolumeSource: corev1.VolumeSource{
+						EmptyDir: &corev1.EmptyDirVolumeSource{},
+					},
+				},
+			},
+			RestartPolicy: corev1.RestartPolicyNever,
+		},
+	}
+
+	newPod := lmevalRec.createPod(job)
+
+	assert.Equal(t, expect, newPod)
+}
+
+func Test_GrpcMTlsPod(t *testing.T) {
+	lmevalRec := LMEvalJobReconciler{
+		Namespace: "test",
+		options: &ServiceOptions{
+			PodImage:         "podimage:latest",
+			DriverImage:      "driver:latest",
+			ImagePullPolicy:  corev1.PullAlways,
+			GrpcPort:         8088,
+			GrpcService:      "grpc-service",
+			GrpcServerSecret: "server-secret",
+			GrpcClientSecret: "client-secret",
+			grpcTLSMode:      TLSMode_mTLS,
+		},
+	}
+	var job = &lmevalservicev1beta1.LMEvalJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test",
+			Namespace: "default",
+			UID:       "for-testing",
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       lmevalservicev1beta1.KindName,
+			APIVersion: lmevalservicev1beta1.Version,
+		},
+		Spec: lmevalservicev1beta1.LMEvalJobSpec{
+			Model: "test",
+			ModelArgs: []lmevalservicev1beta1.Arg{
+				{Name: "arg1", Value: "value1"},
+			},
+			Tasks: []string{"task1", "task2"},
+		},
+	}
+
+	expect := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test",
+			Namespace: "default",
+			Labels: map[string]string{
+				"app.kubernetes.io/name": "fms-lm-eval-service",
+			},
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: lmevalservicev1beta1.Version,
+					Kind:       lmevalservicev1beta1.KindName,
+					Name:       "test",
+					Controller: &isController,
+					UID:        "for-testing",
+				},
+			},
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		Spec: corev1.PodSpec{
+			InitContainers: []corev1.Container{
+				{
+					Name:            "driver",
+					Image:           lmevalRec.options.DriverImage,
+					ImagePullPolicy: lmevalRec.options.ImagePullPolicy,
+					Command:         []string{DriverPath, "--copy", DestDriverPath},
+					SecurityContext: &corev1.SecurityContext{
+						AllowPrivilegeEscalation: &allowPrivilegeEscalation,
+						RunAsUser:                &runAsUser,
+						Capabilities: &corev1.Capabilities{
+							Drop: []corev1.Capability{
+								"ALL",
+							},
+						},
+					},
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							Name:      "shared",
+							MountPath: "/opt/app-root/src/bin",
+						},
+					},
+				},
+			},
+			Containers: []corev1.Container{
+				{
+					Name:            "main",
+					Image:           lmevalRec.options.PodImage,
+					ImagePullPolicy: lmevalRec.options.ImagePullPolicy,
+					Env: []corev1.EnvVar{
+						{
+							Name:  "GRPC_CLIENT_KEY",
+							Value: "/tmp/k8s-grpc-client/certs/tls.key",
```

Comment:
I may have just not found it yet, but where are the key/crt/ca data files coming from here?

Reply:
good observation! The test cases here are to verify the Pod objects. The Pod objects change based on the LMEvaljob objects and the settings of the controller. For example, if the gRPC is using mTLS, the Pod object needs server cert, ca cert, and client key. If there is an EnvSecret in the LMEvalJob object, the Pod object should convert that into the Env list.